### PR TITLE
feat: Add application name to config and connection

### PR DESCRIPTION
**Pull request by strobus**
## Linear Ticket Link

https://linear.app/underlineinfa/issue/SENG-960/🚨-platform-database-issue-report-on-1292024

## Description

This adds `ApplicationName` to the postgres config, and adds it to the database connection. 

Also this adds a `MaxOpenConnections` to the postgers config and sets it on the `sql.DB` object.

## How did you test the changes?

unit tests

## Dependencies

none

### CODE REVIEW

File: generate_address.go
Comment (gjarmstrong) [inline generate_address.go:1]: Should we add the application name to the URL version as well?

**Review by gjarmstrong — APPROVED** [review]
LGTM. I just had one question.

File: generate_address.go
Comment (strobus) [reply generate_address.go:1]: Yeah let me do that. Good catch

**Review by josemarluedke — APPROVED** [review]

File: generate_address.go
Comment (strobus) [reply generate_address.go:1]: I've added `fallback_application_name` to the URL query. Max open connections is not supported on the URL by libpq.

**Pull request merged**